### PR TITLE
feat(native-filters): apply cascading without instant filtering

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Currently enabling cascading in native filters requires instantly applying filters, causing all charts in scope to rerender and retrigger queries. This PR changes child filters to apply the currently selected data mask of the parent filter instead of the currently applied one, removing the need to instantly apply filters.

While making the changes, I considered moving `dataMaskSelected` into Redux instead of adding it as a prop on the chain of components. However, I noticed that many of the intermediate components between the filter tab and `FilterValue` were already using `dataMaskSelected`, which made the benefits of moving it to Redux less appealing. In addition, adding it to Redux would have required either
- introducing a new `dataMaskSelected` property to complement the `dataMask` property (which contains the applied data mask) in the state tree

or 
- refactoring `dataMask` so that it would be split into `dataMask.applied` and `dataMask.selected`

Since both of these would have required fairly significant refactoring, it seems better to perform this refactoring later when the full scope of refactoring needs is clear (e.g. will some of this state be moved into Context or be kept in Redux).

### BFEORE
Currently each time a parent filter is changed, it triggers the filter automatically:
https://user-images.githubusercontent.com/33317356/120623645-962b6000-c468-11eb-86b5-db95319691f4.mp4

### AFTER
Now the parent filter only retriggers the child filters (applying works like before):
https://user-images.githubusercontent.com/33317356/120628894-e2c56a00-c46d-11eb-8200-51d2c73ea10d.mp4

### TESTING INSTRUCTIONS
1. Create a dashboard with cascading filters
2. Make sure the parent filter is *not* set to "Apply changes instantly"
3. Change the selection in the parent filter

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ const Wrapper = styled.div`
 
 type FilterControlsProps = {
   directPathToChild?: string[];
-  dataMaskSelected: DataMaskState;
+  dataMaskSelected: DataMaskStateWithId;
```
Comment (villebro) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx:44]: Bycatch: incorrect datatype

**Comment by michael-s-molina** [comment]
Two issues:
1 - If I select a parent filter then it filters the child filters. However, if I select the child filters first, and select the parent filter after, the values are inconsistent as shown in the video. Is this a problem?
2 - There's an error happening when trying to access the filter after updating the filters

https://user-images.githubusercontent.com/70410625/120641902-974d9480-c44a-11eb-956d-c3df64f39b00.mov

**Review by simcha90 — APPROVED** [review]

**Comment by zhaoyongjie** [comment]
can't remove selected item

https://user-images.githubusercontent.com/2016594/120650854-c139a200-c4b0-11eb-81b8-37a3012226b2.mp4

**Comment by villebro** [comment]
> can't remove selected item

Thanks for flagging this! It seems to be happening on master, too, so isn't directly related to this change.

**Comment by villebro** [comment]
> Two issues:
> 1 - If I select a parent filter then it filters the child filters. However, if I select the child filters first, and select the parent filter after, the values are inconsistent as shown in the video. Is this a problem?
> 2 - There's an error happening when trying to access the filter after updating the filters. This error is not happening on master.

1. I don't think this is a problem, as the value filter supports creating values that don't exist in the query result. But we can potentially consider resetting child values that aren't present when updating the parent. Let's discuss this later.
2. I was able to reproduce this problem on master. Therefore I propose merging this now and following up with a fix for this bug and the bug @zhaoyongjie pointed out (also present on master)

**Pull request merged**